### PR TITLE
llvm: stop running strip-preserve-atime.test

### LINKED ISSUE #284056: Build failure: llvm-16.0.6
### Steps To Reproduce

1. build -A libllvm

### Build log

[9gc2yymj7m5fc0hxxma2afyifbn86yj6-llvm-16.0.6.log](https://gist.github.com/cwp/4468627c2fbc3717a45344d8d1bb5da6)

### Additional context

Discovered this when trying to test #275701 

### Notify maintainers

Dunno who would be appropriate

### Metadata

```console
~> nix-shell -p nix-info --run "nix-info -m"
these 2 paths will be fetched (0.01 MiB download, 0.11 MiB unpacked):
  /nix/store/hrcw8p4cymhyb959w21ha5g1n3dm69n4-DarwinTools-1
  /nix/store/x2v35n7y8ihdwl73dzvjzjlgg4h4lsq9-nix-info
copying path '/nix/store/hrcw8p4cymhyb959w21ha5g1n3dm69n4-DarwinTools-1' from 'https://cache.nixos.org'...
copying path '/nix/store/x2v35n7y8ihdwl73dzvjzjlgg4h4lsq9-nix-info' from 'https://cache.nixos.org'...
 - system: `"aarch64-darwin"`
 - host os: `Darwin 23.1.0, macOS 14.1.2`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.13.3`
 - channels(root): `"nixpkgs"`
 - channels(cwp): `"23.11-23.11-darwin"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`
 ```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by philiptaron**
## Description of changes

The [strip-preserve-atime test](https://raw.githubusercontent.com/llvm/llvm-project/main/llvm/test/tools/llvm-objcopy/ELF/strip-preserve-atime.test) fails in the presence of anti-virus software or other intrusion-detection software that modifies the atime when run.

Fixes #284056; see llvm/llvm-project#82372 for the upstream issue.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by philiptaron** [comment]
I'm building `llvmPackages_{16,17,git}.{clang,llvm,bintools,lld,lldb}` on `x86_64-darwin` and `aarch64-darwin`. Will likely take the better part of a day.

**Review by ghost — COMMENTED** [review]
LGTM
nit: use urls https://github.com/llvm/llvm-project/issues/82372 in comment to make issue lookup easier.

**Review by ghost — APPROVED** [review]

**Comment by philiptaron** [comment]
[This might be a leap year's day bug, or it might be something else](https://gist.github.com/GrahamcOfBorg/06332e8d2617ebc7d7a4f286370a1d6c).

```
error: builder for '/nix/store/jcjpdpx1xicl4hrssxvds3lc8asr4n4f-tzdata-2024a.drv' failed with exit code 2;
       last 10 log lines:
       > now=`./date +%s` && \
       >      future=`expr 28 '*' 366 '*' 24 '*' 60 '*' 60 + $now` && \
       >      ./zdump -i -t $now,$future \
       >              $(find $PWD/check_now.dir/????*/ -type f) \
       >            >check_now.dir/zdump.tab
       > awk \
       >     -v zdump_table=check_now.dir/zdump.tab \
       >       -f checknow.awk zonenow.tab
       > zones Asia/Almaty and Asia/Tashkent identical from now on
       > make: *** [Makefile:944: check_now] Error 1
       For full logs, run 'nix log /nix/store/jcjpdpx1xicl4hrssxvds3lc8asr4n4f-tzdata-2024a.drv'.
```

**Comment by a-n-n-a-l-e-e** [comment]
> [This might be a leap year's day bug, or it might be something else](https://gist.github.com/GrahamcOfBorg/06332e8d2617ebc7d7a4f286370a1d6c).

see https://github.com/NixOS/nixpkgs/pull/292480
started failing Thu Feb 29 18:00:01 UTC 2024

**Comment by a-n-n-a-l-e-e** [comment]
cc lib.teams.llvm: @dtzWill @Ericson2314 @lovek323 @alyssais @RaitoBezarius @rrbutani @sternenseemann

checked build on x64 linux -- passed, unsurprisingly. was planning to merge once get verification that the darwin builds passed but should ping maintainers giving that evail failed due to tzdata bug https://github.com/NixOS/nixpkgs/pull/292480. no modifications related to eval so seems ok to skip here.
```console
GC_INITIAL_HEAP_SIZE=4g nix-env -f =(curl -s https://raw.githubusercontent.com/NixOS/ofborg/released/ofborg/src/outpaths.nix) -qaP --no-name --out-path --argstr path $PWD --arg checkMeta true > out-paths
```
succeeds locally

**Comment by RaitoBezarius** [comment]
There's a PR to fix the tzdata issue, once it's in, let's rebase and verify everything is still fine and go for the merge.

**Comment by philiptaron** [comment]
The tzdata PR is merged. I rebased on staging, and kicked off my quite slow x86_64-darwin build.

**Comment by a-n-n-a-l-e-e** [comment]
> Gentle ping: could this be merged, please?

still waiting for confirmation that the darwin builds succeeded when using antivirus thing.

**Comment by philiptaron** [comment]
> still waiting for confirmation that the darwin builds succeeded when using antivirus thing.

❗ 

I'll get this rebased on staging and building. Did not realize that Raito's message "verify everything is still fine and go for the merge" was on me to provide the ack. 😳

**Comment by appujee** [comment]
Fails frequently on my Macbook (it has antivirus software installed).

**Pull request merged**
